Docs 5646 improve navigability of policies section pa (#140)
mulesoft/docs-api-manager: modules/ROOT/pages/policies-policy-types.adoc

= Policy Types
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: policy, custom, ootb, offline

> Policies are available in differnt options, such as, Default, Custom, and Automated. Each type of the policy differs by its scope, management, and usability.

Reviewer: typo in differnt

Author: Fixed! Thanks!

> == Default Policies

Reviewer: Are we calling them default or provided? What will be the root level name in the navigation bar?

Author: Default Policies. Thanks!

> Mulesoft provides several policies that are ready-to-use and cover different areas, such as authentication, security management, threat protection, and tokenization. You can apply any of these policies using the UI to any of your API endpoints. After you apply a policy, it is managed by the API Manager for its complete lifecycle. You can apply a Default policy to a spcific API. To apply a policy globally to all APIs, you must automate the policy.

Reviewer: technically you can also use the API to apply policies. It would actually be a good thing to add in the future, a section about CI and the link to the API API Manager portal.

Reviewer: typo in spcific

Reviewer: "To apply a policy globally to all APIs, you must automate the policy."

Is it right to say that you automate the policy vs apply a policy that is automatically applied to all policies of your environment? 
Globally is confusing because the scope is not clear, and we want to prevent a reference to global policies (previous name to automated policies that was discarded)

Author: Thanks, Eva! I did not know that context. Made changes accordingly!

> [PA]: DO we release new versions of an existing policy? How does it affect a customer that already has a version applied? What do they have to do with it.

Reviewer: Yes, we do, we should have somewhere a section on versions. 

It doesn't affect the customer, but they will stay with an older version, we should have a section on lifecycle to show how to upgraded to latest versions and promote them to do it.

Author: Yes, thanks! Added a heading to the overview page. Need content to add there.

== Automated Policies

> Automated Policies are applied globally across all APIs. You can globally apply a Default or a Custom policy. However, you require specific roles and permissions to apply Automated policies.

Reviewer: Not globally, to all APIs in an environment.

Reviewer: Are we calling them default or provided?

Can we mention what specific roles and permissions are needed?

Reviewer: it would be good to also add a link to permissions and roles in the see also section.

Author: Default policies! I'll ask @balbifm for more info. Thanks!

Reviewer 2: With what can I help you?

Author: @balbifm  what specific roles and permissions are needed. We need to add a link to the roles and permissions documentation.

== Custom Policies

> Even though Mulesoft provides a wide range of Default policies to suit your business needs, you might have a scenario that may not be covered by a Default policy. In such cases, you can create a Custom policy. You can extend an exiting functionality, or define new ones with a Custom policy. These policies come in two different flavors, namely: Offline and Online.

Reviewer: Default or provided?

Reviewer: typo in exiting

Reviewer: We should have a link to custom policies in the see also section

Author: Default! Addressed others! Thanks!

=== Online Custom Policies
Online Custom policies are applied and managed by the API Manager.

> [PA Need more pertinent information here.]

Reviewer: Online policies are the default and suggested (and highly encouraged) way to apply customers since this is done via API Manager. It allows, amongst other things, to leverage the platform to handle a policy lifecycle, keep the applied policies in sync with API Manager, to be protected by the Gatekeeper mechanism and to do client enforcement. 

@balbifm could you confirm whether client validation is enabled for offline?

Reviewer 2: It is, just the runtime needs to be paired with APIM

Author: Done! Thanks!

=== Offline Custom Policies

> Offline Custom policies are applied directly to the runtime. You do need to apply them using the API Manager. 

Reviewer: We have a section for offline custom policies: https://docs.mulesoft.com/api-manager/2.x/policies-custom-offline-landing-page

Author: Added a link! Thanks!

[PA Need more pertinent information here.]

> == See Also

Reviewer: We are missing links here: 
- Offline policies documentation
- Custom policies documentation
- Automated policies documentation
- Provided policies list

Author: Added! Thanks! Priya

= Policy Types
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: policy, custom, ootb, offline

> Policies are available in different options, such as, Default, Custom, and Automated. Each type of the policy differs by its scope, management, and usability.

Reviewer 3: ```suggestion
Policies can be default, custom, or automated. Each type of policy has its own scope, management, and usability.
```

Author: Done! Thanks!

== Default Policies

> Mulesoft provides several policies that are ready-to-use and cover different areas, such as authentication, security management, threat protection, and tokenization. You can apply any of these policies using the UI to any of your API endpoints. You can also use an API to apply policies.

Reviewer 3: GLOBALLY SEARCH FOR AND FIX to "MuleSoft"

Reviewer 3: ```suggestion
MuleSoft provides several ready-to-use policies for areas such as authentication, security management, threat protection, and tokenization. Using the UI, you can apply any of these policies to any of your API endpoints. You can also use an API to apply policies.
```

Author: Done! Thanks!

> Mulesoft provides several policies that are ready-to-use and cover different areas, such as authentication, security management, threat protection, and tokenization. You can apply any of these policies using the UI to any of your API endpoints. You can also use an API to apply policies.

Reviewer: More than using the UI, shouldn't it be: Using Anypoint platform? (this should be either UI or API)

Author: changed it to "You can apply any of these policies to any of your API endpoints using API Manager in Anypoint Platform." Done! Thanks!

> After you apply a policy, it is managed by the API Manager for its complete lifecycle. You can apply a Default policy to a specific API. To automatically apply a policy to to all APIs in your environment, you must automate the policy.

Reviewer: I don't think automate the policy applies in this scenario, I would change it for "you must configure it as an automated policy"

Author: Done! Thanks!

> After you apply a policy, it is managed by the API Manager for its complete lifecycle. You can apply a Default policy to a specific API. To automatically apply a policy to to all APIs in your environment, you must configure it as  a policy.

Reviewer 3: ```suggestion
After you apply a policy, it is managed for its complete lifecycle by API Manager. You can apply a default policy to a specific API. To automatically apply a policy to all APIs in your environment, you must configure it as a policy.
```
The last sentence does not make sense. What kind of policy? Did you mean "automatic"?

Reviewer: Agree, it should be "you must configure it as an automated policy"

Author: Done! Thanks!

//It would actually be a good thing to add in the future, a section about CI and the link to the API API Manager portal. -- Eva?

//[PA]: DO we release new versions of an existing policy? How does it affect a customer that already has a version applied? What do they have to do with it.

> == Automated Policies

Reviewer 3: This is referred to earlier as "automatic." GLOBALLY standardize to the proper name.

Reviewer: It should be automated, not automatic.

Author: It is automated everywhere. Done! Thanks!

> You can apply any Default or a Custom policy to all the APIs in your environment, which is then referred to as Automated Policies. However, you require specific roles and permissions to apply Automated policies.

Reviewer 3: ```suggestion
You can apply any default or a custom policy to all the APIs in your environment, making that policy an _automated policy_. However, you require specific access to create automated policies.
```

Author: Done! Thanks!

> // what specific roles and permissions are needed. Ask Fede? Add a link to the roles and permissions documentation.

Reviewer: Ask Tobias Lichtig, Sergio Behrends or Tomás Bruno.

Author: Done! Thanks!

== Custom Policies

> Even though Mulesoft provides a wide range of Default policies to suit your business needs, you might have a scenario that may not be covered by a Default policy. In such cases, you can create a Custom policy. You can extend an existing functionality, or define new ones with a Custom policy. These policies come in two different flavors, namely: Offline and Online.

Reviewer 3: ```suggestion
Although MuleSoft provides you with several ready-to-use default policies, you might want to create a custom policy to meet your specific business needs. You can either customize an existing policy, or you can create an entirely new custom policy. These policies are categorized as either online or offline.
```

Author: Done! Thanks!

=== Online Custom Policies

> Online Custom policies are applied and managed by the API Manager, which is the default and recommended way to apply policies. Online policies enable you to manage the policy lifecycle by leveraging the platform. This way, the applied policies are always in sync with API Manager, and are protected by the Gatekeeper mechanism to perform client enforcement.

Reviewer 3: ```suggestion
Online Custom policies are applied and managed by API Manager, which is the default and recommended way to apply policies. Online policies enable you to manage the policy lifecycle by leveraging Anypoint Platform. This way, the applied policies are always in sync with API Manager, and are protected by the gatekeeper mechanism to perform client enforcement.
```

Reviewer: In the last sentence: "and are protected by the gatekeeper mechanism at startup"

Author: Done! Thanks!

> To perform client validations, you must pair the Mule Runtime with API Manager.

Reviewer 3: ```suggestion
To perform client validations, you must pair Mule with API Manager.
```

Reviewer: This is actually incorrect, we should remove this sentence from here.

Author: Done! Thanks!

//How can they do this--Fede Balbi?

=== Offline Custom Policies

> Offline Custom policies are applied directly to the runtime. You do need to apply them using the API Manager. However, you manage the policy lifecycle manually with this option. Because of the chance that a policy might be out-of-sync with API Manager, this option is not recommended. 

Reviewer 3: ```suggestion
Offline custom policies are applied directly to the runtime and managed manually. Because they can become out of sync with API Manager, using this type of policy is not recommended. 
```

Author: Done! Thanks!

> // Does this look fine? Anything else to add here? Eva, defe balbi?

Reviewer: Yes, two more things:

1) Offline policies are not protected by the gatekeeper mechanism at startup. 
2) Client credentials validation is not supported in offline policies.

Author: Done! Thanks!

== See Also

> * xref:policies-mule3.adoc[Policies in Mule 3]

Reviewer: We should have a link to gatekeeper documentation (is mentioned many times)

Author: Added! Thanks! Done!
